### ISSUE TITLE
[Improvement] Add HEAD support

### ISSUE THREAD

**Issue by simonetognolo** [GitTools/GitVersion#3233]
Actually this project doesn't support to get version from HEAD commit. GetVersionTask v5.3.7 supports it, and I'm stuck with it because I've submodules in my project.

## Detailed Description

- Add an option to allow detached branches (default is false);
- Handle the initial check `EnsureHeadIsNotDetached` according to previous option;
- Get the closest tag to current commit (according to previous option).

## Context

- It'll be possible to get version from HEAD commit;
- Compile of git project at HEAD commit wouldn't end with error;
- Actual Gitversion already supports subversion;
- Fix when a commit is detached but has a tag (actually doesn't work because `EnsureHeadIsNotDetached` check prevent to look for commit info).

## Possible Implementation

In `RepositoryStore.GetCurrentCommitSemanticVersions` you can check if current commit is tagged, or get the closest tag to the current commit.

### RepositoryStore.cs

```cs
private IEnumerable<SemanticVersion> GetCurrentCommitSemanticVersions(ICommit? commit, string? tagPrefix, ITag tag, bool getClosestTag = false)
{
    var targetCommit = tag.PeeledTargetCommit();
    if (targetCommit == null)
        return Array.Empty<SemanticVersion>();

    var commitToCompare = getClosestTag ? FindMergeBase(commit, targetCommit) : commit;

    var tagName = tag.Name.Friendly;

    return Equals(targetCommit, commitToCompare) && SemanticVersion.TryParse(tagName, tagPrefix, out var version)
        ? new[] { version }
        : Array.Empty<SemanticVersion>();
}
```

**Comment by asbjornu** [comment]
Can you please submit a PR with this change so we can get a grasp of whether it's breaking any tests, and if so, how many?

**Comment by simonetognolo** [comment]
> Can you please submit a PR with this change so we can get a grasp of whether it's breaking any tests, and if so, how many?

I've created the branch locally, but I've no rights to push & create pull request.

**Comment by gep13** [comment]
> @simonetognolo said...
I've created the branch locally, but I've no rights to push & create pull request.

Have you created a fork of the repository on GitHub? If not, do that, and push your branch to your fork. Once that is done, you should be able to create a Pull Request.

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request

**Comment by asbjornu** [comment]
Superseded by #3338.
